### PR TITLE
Add weight and scale to hopper simulator

### PR DESCRIPTION
**Pull request by rsinnet**
The hopper controller can now track these and publishes weight and empty
bin information. Also includes refilling and tests.

### CODE REVIEW

**Comment by rsinnet** [comment]
Going to refactor.

File: miso_peripherals/test/test_sim_hopper.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    Bool,
+    Float32,
+)
+
+TIMEOUT = 1
+
+
+def empty(slot):
+    return rospy.wait_for_message(
+        '/hopper/{:s}/empty'.format(slot.value),
```
Comment (ludwig) [inline miso_peripherals/test/test_sim_hopper.py:27]: Doesn't `{}` already do string formatting by default?

File: miso_peripherals/src/miso_peripherals/hopper/ram_driver.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"""A controller for a RAM hopper."""
```
Comment (ludwig) [inline miso_peripherals/src/miso_peripherals/hopper/ram_driver.py:1]: Out of curiosity, what does the RAM acronym stand for? None of the files here seem to explain or expand it out.

Comment (rsinnet) [reply miso_peripherals/src/miso_peripherals/hopper/ram_driver.py:1]: 😆 Not sure, but it's the name of the company: https://www.autoequipllc.com/ram-280f/

File: miso_peripherals/test/test_sim_hopper.py (hunk 1 above)
Comment (rsinnet) [reply miso_peripherals/test/test_sim_hopper.py:27]: Oh that's cool.

Comment (ludwig) [reply miso_peripherals/test/test_sim_hopper.py:27]: To get the repr, you need to do something like `{!r}`

Comment (rsinnet) [reply miso_peripherals/test/test_sim_hopper.py]: Fixed.

**Pull request merged**